adr/0009-translations-2.0.md

```diff
@@ -0,0 +1,95 @@
+- Change `frontend/get_translations` to allow specifying area that you are interested in
+- We need to start using references and write specific translations down once. Right now each integration defines `already_configured` etc. We use this syntax in the frontend: `[%key:state::binary_sensor::battery::off%]`.
+- Update scaffold scripts to use references.
+- Start using `state` in `strings.json`. It is going to be a dictionary that has a `default` entry and one entry per device class. Just like we have now in the frontend.
+    ```json
+    {
+      "state": {
+        "binary_sensor": {
+          "default": {
+            "off": "[%key:state::default::off%]",
+            "on": "[%key:state::default::on%]"
+          },
+          "battery": {
+            "off": "Normal",
+            "on": "Low"
+          },
+        }
+      }
+    }
+    ```
+- Import all existing state keys + translations from frontend
+- Add a new `state_attributes` to `strings.json` to allow offering translations for state attributes.
+  ```json
+  {
+    "state_attributes": {
+      "climate": {
+        "hvac_action": {
+          "off": "Off",
+          "heating": "Heating",
+          "cooling": "Cooling",
+          "drying": "Drying",
+          "idle": "Idle",
+          "fan": "Fan"
+        }
+      }
+    }
+  }
+  ```
+- Import all existing state attribute keys + translations from frontend
+- Replace as many existing translations as possible with references
+
+#### Frontend
+Load part of backend translations when necessary.
+
+- On page load, load `state` only.
+- Fetch integration titles when opening config or dev tools panels
+- Fetch config flow translations when config flow started (not handler picked)
+- Fetch options flow translations when option flow started
+- Fetch device automation strings when opening script/automation editor
+- Remove built-in translations of domains (`domain.*`) and use backend provided strings instead.
+- Remove built-in translations for states (`state`) and use backend provided strings instead.
+- Remove built-in translations for state attributes (`state_attributes`) and use backend provided strings instead.
+- Adopt cleaning keys script from backend ([see this PR](https://github.com/home-assistant/core/pull/33802))
+- Remove all state and state attributes keys from translations once imported into backend
+
+## Consequences
+
+- Frontend will load faster and use less memory
+- Couple translations of integrations to the integration instead of the frontend release
+
+The `moon` and `season` integrations will no longer have their entities translated. I suggest that if we see a need to bring this back, we could consider migrating `moon` and `season` to not write sensor entities but instead have them write states under their own domain (`moon.moon`) like the `sun` integration does. They can then add `state` to their own `strings.json` which can then be loaded when the frontend loads.
```

Comment:
As user of moon and season, I agree, that are no sensors and we should migrate this as part of this ADR change and not suggest it. So we have `sun`, `moon`, `season` which also should provide later device automation and condition. That stuff is really useful for automation (okay, moon is less as season).